dlrover/python/diagnosis/common/diagnosis_data.py

```diff
@@ -11,69 +11,117 @@
-    def __init__(self):
-        pass
-
-    @abstractmethod
-    def get_timestamp(self) -> float:
-        pass
-
-    @abstractmethod
-    def get_type(self) -> str:
-        pass
-
-
-class CudaLog(DiagnosisData):
-    def __init__(self, timestamp: int):
+    def __init__(self, data_type, timestamp: int = 0, data_content: str = ""):
+        self._data_type = data_type
         if timestamp == 0:
-            self.timestamp = int(round(datetime.now().timestamp()))
+            self._timestamp = int(round(datetime.now().timestamp()))
         else:
-            self.timestamp = timestamp
-
-    def get_timestamp(self) -> int:
-        return self.timestamp
-
-    def get_type(self) -> str:
-        return DiagnosisDataType.CUDALOG
-
-
-class TrainingLog(DiagnosisData):
+            self._timestamp = timestamp
+        self._data_content = data_content
+
+    @property
+    def data_type(self) -> str:
+        return self._data_type
+
+    @property
+    def timestamp(self) -> int:
+        return self._timestamp
+
+    @property
+    def data_content(self) -> str:
+        return self._data_content
+
+
+class AgentMetric(DiagnosisData):
+    def __init__(
+        self,
+        timestamp: int = 0,
+        data_type: str = DiagnosisDataType.GENERIC,
+        data_content: str = "",
+        is_final_result=False,
+        need_report=False,
+        node_id: int = -1,
+        node_type: str = "",
+        node_rank: int = -1,
+    ):
+        """
+        General metric
+
+        Args:
+            data_type (str): Type of metric. Defaults to "GENERIC".
+            data_content (str): Content of the metric. Defaults to "".
+            is_final_result (bool, optional): Whether the metric is final
+                result or not. Defaults to False.
+            need_report (bool, optional): Whether the metric needs
+                report(to Brain). Defaults to False.
+            node_id (int): Node ID. Defaults to -1.
+            node_type (str): Node type. Defaults to "".
+            node_rank (int): Node rank. Defaults to -1.
+        """
+
+        super().__init__(data_type, timestamp, data_content)
+        self._is_final_result = is_final_result
+        self._need_report = need_report
+        self._node_id = node_id
+        self._node_type = node_type
+        self._node_rank = node_rank
+
+    @property
+    def is_final_result(self):
+        return self._is_final_result
+
+    @property
+    def need_report(self):
+        return self._need_report
+
+    @property
+    def node_id(self):
+        return self._node_id
+
+    @property
+    def node_type(self):
+        return self._node_type
+
+    @property
+    def node_rank(self):
+        return self._node_rank
+
+    def is_resolvable(self):
+        if self.data_type == DiagnosisDataType.TRAINING_HANG_DETECTION:
+            return True
+        # TODO: add more resolvable metric type later
+        return False
+
+
+class TrainingLog(AgentMetric):
```

Comment:
I am a little confused here. Why do AgentMetrics inherit DiagnosisData and TrainingLog inherit AgentMetrics? I fail to catch up with the design of AgentMetrics here.

Reply:
Refactored.